core/app.cc

```diff
@@ -394,25 +406,42 @@ void application::prepare_argv(elf::program *program)
     }
     contig_argv[_args.size() + 1 + envcount] = nullptr;
 
-    Elf64_auxv_t* _auxv =
-        reinterpret_cast<Elf64_auxv_t *>(&contig_argv[_args.size() + 1 + envcount + 1]);
-    int auxv_idx = 0;
+    _auxv = reinterpret_cast<Elf64_auxv_t *>(&contig_argv[_args.size() + 1 + envcount + 1]);
+    _auxv_idx = 0;
 
     // Pass the VDSO library to the application.
-    _auxv[auxv_idx].a_type = AT_SYSINFO_EHDR;
-    _auxv[auxv_idx++].a_un.a_val = reinterpret_cast<uint64_t>(program->get_libvdso_base());
+    _auxv[_auxv_idx].a_type = AT_SYSINFO_EHDR;
+    _auxv[_auxv_idx++].a_un.a_val = reinterpret_cast<uint64_t>(program->get_libvdso_base());
+
+    _auxv[_auxv_idx].a_type = AT_PAGESZ;
+    _auxv[_auxv_idx++].a_un.a_val = sysconf(_SC_PAGESIZE);
+
+    _auxv[_auxv_idx].a_type = AT_MINSIGSTKSZ;
+    _auxv[_auxv_idx++].a_un.a_val = sysconf(_SC_MINSIGSTKSZ);
+
+    _auxv[_auxv_idx].a_type = AT_RANDOM;
+    _auxv[_auxv_idx++].a_un.a_val = reinterpret_cast<uint64_t>(random_bytes);
+
+    _auxv[_auxv_idx].a_type = AT_NULL;
+    _auxv[_auxv_idx].a_un.a_val = 0;
+}
+
+void application::augment_auxv()
+{
+    _auxv[_auxv_idx].a_type = AT_PHDR;
```

Comment:
Can you maybe assert() that there is enough room for the 4 items you want to add, from _auxv_idx until max_auxv_parameters_count?
Otherwise (e.g., if this function is called twice) we can have silent corruption.

Reply:
Addressed by better comments in the latest version of this PR.